### PR TITLE
Fix _.get

### PR DESCRIPTION
**Pull request by clujin**
This PR addresses 2 issues with `_.get` as reported [here](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore/issues/178).

### CODE REVIEW

**Comment by stevemao** [comment]
This looks a bit tricky. Could you add unit tests https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore/blob/master/tests/unit/all.js?

**Comment by alex-page** [comment]
@cylim @stevemao @clujin This is failing when the object has an array as the child:
```
var object = { a: { b: { c: [ 3 ] } } };
var result = Get( object2, 'a.b.c[ 0 ]', 1 );  // returns 1
```

It is also doing some crazy stuff when the split key strings are multiple characters:
```
var object = { abc: { def: { ghi: [ 3 ] } } };
var result = Get( object2, 'abc.def.ghi[ 0 ]', 1 );  // returns 1
```

**Comment by stevemao** [comment]
@clujin could you address @alex-page's comment please? Thanks.

File: README.md [hunk 1]
```diff
@@ -1836,8 +1836,8 @@ Gets the value at path of object.
   // output: 3
   
   // Native
-  const get = (obj, path, defaultValue) => path.split(".")
-  .reduce((a, c) => (a && a[c] ? a[c] : (defaultValue || null)), obj)
+  const get = (obj, path, defaultValue) => String.prototype.split.call(path, /,?[\[\]]?\.?/)
```
Comment (dartandrevinsky) [inline README.md]: `[\[` better be replaced with `[[` since escaping of `[` in a `[...]` construct is useless

Comment (andrevinsky) [reply README.md]: You haven't tested it on path keys longer than 1 character, have you? (@clujin)

**Comment by clujin** [comment]
Thank you everyone for the feedback. I've just updated the PR with the following changes:

- Add default parameter value for `defaultValue` to avoid a bug when it is falsey.
- Simplify regex to use a single set for splitting. It will be an optional greedy approach to work nicely with longer than 1 character string keys.
- Add a Boolean filter to remove empty strings `''` that are produced from the single set splitting (ie. `].` cases) as well as from the special cases ending in `]` (ie. `a[0][1]`).

**Comment by cht8687** [comment]
thanks for the PR! @clujin

**Comment by bmpGames** [comment]
Now _.get is broken :(
Because we are trying to get property of defaultValue in a next iteration

```
const get = (obj, path, defaultValue = null) =>
  String.prototype.split.call(path, /[,[\].]+?/)
    .filter(Boolean)
    .reduce((a, c) => (Object.hasOwnProperty.call(a,c) ? a[c] : defaultValue), obj)

var object = { a: [{ b: { c: 3 } }] };
var result = get(object, 'a[0].c.b'); 
// Uncaught TypeError: Cannot convert undefined or null to object at hasOwnProperty
```

**Comment by stevemao** [comment]
@bmpGames Could you send a PR?

**Comment by bmpGames** [comment]
@stevemao Ok, I'll do it a bit later

**Comment by bmpGames** [comment]
@stevemao Done #205 :)

**Pull request merged**